test/runtests.jl

```diff
@@ -14,33 +14,24 @@ end
 
 args = Base.shell_split(get(ENV, "JULIA_MPIEXEC_TEST_ARGS", ""))
 
-function runtests()
-    nprocs = clamp(Sys.CPU_THREADS, 2, 4)
-    testdir = dirname(@__FILE__)
-    istest(f) = endswith(f, ".jl") && startswith(f, "test_")
-    testfiles = sort(filter(istest, readdir(testdir)))
+nprocs = clamp(Sys.CPU_THREADS, 2, 4)
+testdir = @__DIR__
+istest(f) = endswith(f, ".jl") && startswith(f, "test_")
+testfiles = sort(filter(istest, readdir(testdir)))
 
-    nfail = 0
-    printstyled("Running MPI.jl tests\n"; color=:white)
-    
-    for f in testfiles
-        mpiexec() do cmd
-            cmd = `$cmd $args`
-            if f == "test_spawn.jl"
-                run(`$cmd -n 1 $(Base.julia_cmd()) $(joinpath(testdir, f))`)
-            elseif f == "test_threads.jl"
-                withenv("JULIA_NUM_THREAD" => "4") do
-                    run(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`)
-                end
-            else
-                run(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`)
+@testset "$f" for f in testfiles
+    mpiexec() do cmd
+        cmd = `$cmd $args`
+        p = if f == "test_spawn.jl"
+            run(`$cmd -n 1 $(Base.julia_cmd()) $(joinpath(testdir, f))`; wait = true)
+        elseif f == "test_threads.jl"
+            withenv("JULIA_NUM_THREAD" => "4") do
+                run(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`; wait = true)
             end
+        else
+            run(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`; wait = true)
         end
-        Base.with_output_color(:green,stdout) do io
-            println(io,"\tSUCCESS: $f")
-        end
+        wait(p)
```

Comment:
isn't this redundant with `wait=true`?

Reply:
In the REPL ``run(`...`; wait=true)`` returns immediately, but in a script it does seem to wait without this.  But see my comment below about not running the processes asynchronously

Reply:
> In the REPL `` run(`...`; wait=true) `` returns immediately

I can't replicate this:
```
julia> @time run(`sleep 3`; wait=true)
  3.003904 seconds (50 allocations: 1.906 KiB)
Process(`sleep 3`, ProcessExited(0))
```

Reply:
I looked at my REPL history: I was testing
```julia
run(`sleep 10`; wait = false)
```
:expressionless: